Mathlib/Data/Finset/Basic.lean

```diff
@@ -1131,6 +1122,8 @@ theorem eq_of_not_mem_of_mem_insert (ha : b ∈ insert a s) (hb : b ∉ s) : b =
   (mem_insert.1 ha).resolve_right hb
 #align finset.eq_of_not_mem_of_mem_insert Finset.eq_of_not_mem_of_mem_insert
 
+@[simp] lemma insert_empty : insert a (∅ : Finset α) = {a} := rfl
```

Comment:
This already exists as `IsLawfulSingleton.insert_emptyc_eq`, which I assume is exported globally.

Reply:
Yes but it's not simp nor provable by `rfl`

Reply:
```suggestion
-- A version of `IsLawfulSingleton.insert_emptyc_eq` that works with `dsimp`
@[simp] lemma insert_empty : insert a (∅ : Finset α) = {a} := rfl
```